src/client/datascience/interactive-ipynb/nativeEditor.ts

```diff
@@ -233,6 +236,58 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
         this.previouslyNotTrusted = !this._model.isTrusted;
     }
 
+    @captureTelemetry(Telemetry.SyncAllCells)
+    public async syncAllCells(): Promise<void> {
+        const timer = new StopWatch();
+        // Ask the UI for all of our all cells
+        this.waitingForMessageResponse = {
+            message: InteractiveWindowMessages.ReturnAllCellCode,
+            result: createDeferred<string[]>()
+        };
+        await this.postMessage(InteractiveWindowMessages.GetAllCellCode);
+        const code = await this.waitingForMessageResponse.result.promise;
+
+        // Upload these cells into our model (so they match what is in the UI)
+        this.model.replaceCells(
+            this.model.cells.map((c, i) => {
+                return {
+                    ...c,
+                    data: {
+                        ...c.data,
+                        source: code[i]
+                    }
+                    // tslint:disable-next-line: no-any
+                } as any; // Deal with nyc problems
+            })
+        );
+        traceInfo(`Sync all elapsed type: ${timer.elapsedTime}`);
+    }
+
+    @captureTelemetry(Telemetry.SyncSingleCell)
+    public async syncCell(cellId: string): Promise<void> {
+        // Ask the UI for the code for this cell
+        this.waitingForMessageResponse = {
+            message: InteractiveWindowMessages.ReturnCellCode,
+            result: createDeferred<ICell | undefined>()
+        };
+        await this.postMessage(InteractiveWindowMessages.GetCellCode, cellId);
+        const code = await this.waitingForMessageResponse.result.promise;
```

Comment:
This will prevent the `syncAllCells` from working, if this method is call after `syncAlLCells`
Meaning we'd end up only saving the specific cell.